id: [0, 0, 0, 0]
# 4个port分别对应LF RF LB RB
port_name: ["/dev/ttyUSB4","/dev/ttyUSB5","/dev/ttyUSB6","/dev/ttyUSB7"]  # 串口控制权限在/etc/udev/rules.d/70-ttyusb.rules中
reduction_ratio: 6.33   # 减速比

cmd_type: 0 # 0：键盘远程控制，1：轨迹跟踪，2：速度控制，3：贝塞尔轨迹规划，4：五次多项式轨迹规划

motor_ctrl_data:
  mode: 1   # 控制方式：刹车==0，FOC闭环==1，电机标定==2
  ctrl_frequency: 500   # 控制频率

  #FOC闭环控制参数：实际给FOC的指令力矩为：K_P*delta_Pos + K_W*delta_W + T
  pos_kp_kw: [0.6, 0.1] # 位置模式下的关节刚度、速度系数。PD控制，kp为比例系数，kw为微分系数
  vel_kp_kw: [0.0, 0.02] # 速度模式下的关节刚度、速度系数。P控制，kp需为0，kw为比例系数
  trq_kp_kw: [0.0, 0.0] # 力矩模式下的关节刚度、速度系数，均为0，开环
  traj_kp_kw: [1.5, 0.05] # 轨迹规划模式下的关节刚度、速度系数。PD控制，kp为比例系数，kw为微分系数(kp太小时位置误差较大，kw太高会疯狂震动)

  #目标角度：°    目标速度：rad/s   目标力矩：NM
  goal_pos_vec: [20, -20, 20, -20] 
  goal_vel: 10
  goal_trq: 0.2
  goal_traj: [90, -90, 90, -90]  # 路点

  interpolation:
    total_run_time: 3   # 轨迹总运行时间